src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/INodeWithDebugInfo.cs

```diff
@@ -19,12 +23,37 @@ public DebugLocInfo(int nativeOffset, string fileName, int lineNumber, int colNu
             ColNumber = colNumber;
         }
     }
-
+    
     public interface INodeWithDebugInfo
     {
         DebugLocInfo[] DebugLocInfos
         {
             get;
         }
+
+        DebugVarInfo[] DebugVarInfos
+        {
+            get;
+        }
+    }
+
+     
+    public struct DebugVarInfo
+    {
+        public readonly uint VarNumber; // For parameters, this is the parameter index.
+                                        // For local variables, this is the slot number.
```

Comment:
The comment is not really true. For local, VarNumber = slot number + # of parameters.
In fact, I don't think this is needed for Object emission since we've already resolved name based on this. 

NativeVarInfo already has this VarNumber and I'm tempted to delete it too, but then we need to define/teach new NativeVarInfo which may be more disruptive.